src/main/java/ua/vedroid/cinema/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package ua.vedroid.cinema.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import ua.vedroid.cinema.dao.ShoppingCartDao;
+import ua.vedroid.cinema.dao.TicketDao;
+import ua.vedroid.cinema.lib.Inject;
+import ua.vedroid.cinema.lib.Service;
+import ua.vedroid.cinema.model.MovieSession;
+import ua.vedroid.cinema.model.ShoppingCart;
+import ua.vedroid.cinema.model.Ticket;
+import ua.vedroid.cinema.model.User;
+import ua.vedroid.cinema.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        return shoppingCartDao.add(shoppingCart);
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        return shoppingCartDao.getAll();
+    }
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
```

Comment:
if there is no shopping cart for the user - we still save ticket to the db, is it ok?